server/src/main/java/org/elasticsearch/snapshots/Snapshot.java

```diff
@@ -38,9 +40,9 @@
     /**
      * Constructs a snapshot.
      */
-    public Snapshot(final String repository, final SnapshotId snapshotId) {
+    public Snapshot(String repository, @Nullable SnapshotId snapshotId) {
```

Comment:
This is kinda dirty admittedly, but it seems like it's the shortest route to piggy back onto `SnapshotDeletionsInProgress.Entry`. Unfortunately that class doesn't have a reference to the `repository` otherwise so if I encode this as anything but a `null` `SnapshotId` here I'd have to add the repository to the entry and have it here redundantly.